### PR TITLE
Add methods to `Descriptor` + `DescriptorPublicKey`

### PR DESCRIPTION
**Pull request by reez**
### Description

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.DescriptorPublicKey.html#method.master_fingerprint

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.DescriptorPublicKey.html#method.is_multipath

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.Descriptor.html#method.is_multipath

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.Descriptor.html#method.into_single_descriptors

**_Would love for this to get into our beta6_**

### Notes to the reviewers

Use Cases:
- creating a wallet from a descriptor having the ability to check if it's multipath or not
- creating a wallet from an xpub having the ability to construct a DescriptorPublicKey fromString then get the fingerprint

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [x] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by rustaceanrob** [comment]
Concept ACK

Cool additions

File: bdk-ffi/src/descriptor.rs [hunk 1]
```diff
@@ -266,6 +266,10 @@ impl Descriptor {
         let key_map = &self.key_map;
         descriptor.to_string_with_secret(key_map)
     }
+
+    pub(crate) fn is_multipath(&self) -> bool {
```
Comment (rustaceanrob) [inline bdk-ffi/src/descriptor.rs:270]: Just out of curiosity, why are all of these access modifiers `crate` level? Clearly has no effect on the bindings but seems like a scope definition that doesn't actually apply

Comment (reez) [reply bdk-ffi/src/descriptor.rs:272]: good question/point-

**Comment by reez** [comment]
> Concept ACK
> 
> Cool additions

Thanks!

**Comment by reez** [comment]
I get a failure because `cargo clippy` doesn't like the naming convention, I think the best option to get this to pass are renaming:

```
    pub(crate) fn to_single_descriptors(&self) -> Result<Vec<Arc<Descriptor>>, MiniscriptError> {
    }
```

will do that right now just cuz I would like this to pass even though that naming isn't exactly equivalent to the function we are wrapping.

File: bdk-ffi/src/bdk.udl [hunk 2]
```diff
@@ -549,6 +590,12 @@ interface DescriptorPublicKey {
   DescriptorPublicKey extend([ByRef] DerivationPath path);
 
   string as_string();
+
+/// Whether or not this key has multiple derivation paths.
```
[the remark this answers is not in the document]
Comment (reez) [reply bdk-ffi/src/bdk.udl:594]: yep sorry didn't update after comment on https://github.com/bitcoindevkit/bdk-ffi/pull/615#pullrequestreview-2437116595 but will update now-

**Comment by reez** [comment]
> Just a tiny comment on the identation of the docstrings, but otherwise this is good to go IMO.

Updated docstrings indentation, rebased, ready to merge 👍

**Review by thunderbiscuit — APPROVED** [review]

**Comment by thunderbiscuit** [comment]
Sorry man I merged #629 before this one, so you'll have to rebase, but feel free to merge afterwards!

**Pull request merged**